### DISCUSSION TITLE
【柴柴教學】免費實用小工具- AI圖片無損放大(Image Upscalers)

### DISCUSSION THREAD

**Discussion by wolfyeva** [1101-GS3073-C/Python-Community-GS3073-C#306] [General]
# AI圖片無損放大(Image Upscalers)
## 問題
現在很多網站、軟體如instagram、Messenger，在上傳檔案的時候都會自動降低解析度，在得不到原圖的時候，可以試著使用AI技術將圖片無損放大，也就是想辦法增加其解析度。
### [生成對抗網路(GAN)](https://zh.wikipedia.org/wiki/%E7%94%9F%E6%88%90%E5%AF%B9%E6%8A%97%E7%BD%91%E7%BB%9C)
生成對抗網路(Generative Adversarial Network)是非監督式學習的一種方法，通過讓兩個神經網路相互博弈的方式進行學習。常見的應用為生成**以假亂真的圖片**。

目前最常見實行圖片無損放大功能的AI模型，都是使用生成對抗網路(GAN)實作，大致上有以下幾種：
- [SRGAN](https://github.com/tensorlayer/srgan) (Super Resolution GAN)
- [ESRGAN](https://github.com/xinntao/ESRGAN) (Enhanced SRGAN)

### 比較
![image](https://user-images.githubusercontent.com/81672797/165902420-f5da1578-6573-462f-83b7-dc92574fcca9.png)
> 這個例子將照片先模糊化，再分別用SRGAN和ESRGAN放大圖片。

結果：可見ESRGAN比SRGAN出色(更像真實照片)。

#### 1. 登入並上傳照片
![螢幕擷取畫面 2022-04-29 150520](https://user-images.githubusercontent.com/81672797/165900184-5106b9f4-0ff3-45af-a253-29514ad18fbb.png)

#### 2. 設定
可以選擇：
1. Processing (圖片的內容)：會根據選擇不同而使用不同的AI模型進行圖片放大
2. Scale (要放幾倍大)
3. Effects (是否加強顏色)
![螢幕擷取畫面 2022-04-29 150257](https://user-images.githubusercontent.com/81672797/165900440-af81204e-80ea-413b-8446-27387f036ab1.png)
#### 3. 完成 & 比較
> 可明顯看出圖片線條邊緣減少鋸齒狀的感覺！
![螢幕擷取畫面 2022-04-29 150442](https://user-images.githubusercontent.com/81672797/165900796-2b4734e6-7e05-4377-9486-6169b23622c9.png)

### 其他免費實作網站
- [BIGJPG](https://bigjpg.com/tw)
- [SMART UPSCALER](https://icons8.com/upscaler)

**Comment by lx3211** [comment]
哇！差別好大！謝謝分享

**Comment by xCk27x** [comment]
挖賽 超好用的 !
但目前看來缺點是影像處理的耗時比較久
甚至還有選項是放大完成時寄email給我XD
更: 最後花了20分鐘終於收到email了==

**Comment by xCk27x** [comment]
試著把板書的照片進行放大
效果很不錯
粉筆的線條變的清晰了許多
>已經加入我的最愛💕

但考慮到照片轉換需要花的時間...

**Comment by limitx0** [comment]
超實用的網站
看到畫質差的圖片就很難受
